packages/nimble-components/src/table/types.ts

```diff
@@ -25,6 +25,18 @@ export interface TableRecord {
     [key: TableFieldName]: TableFieldValue;
 }
 
+export type StringField<FieldName extends string> = {
+    [name in FieldName]: string | null | undefined;
+};
+
+export interface TableCellState<
+    TCellRecord extends TableRecord,
```

Comment:
while this makes sense from a type perspective I think it's confusing from a naming / concept perspective.

Just language-wise a Table Record instance will have fields defined by the user's table data property while a Cell Record has fields defined by a plugin.

Maybe we should have:
```ts

// Maybe we don't actually export this type, conceptually I can't think of when I would want to intentionally mixup TableDataRecords and TableCellRecords
interface TableRecord {
    [key: FieldName]: FieldValue;
}

/** Can give a nice doc comment about how the fields are completely chosen by user's and the column configuration maps to stuff */
export interface TableDataRecord extends TableRecord {}

/** Can give a nice doc comment about how CellRecords fields are defined by plugins and used for plugin templates and map to the table's data */
export interface TableCellRecord extends TableRecord {}
```